e2e/test/scenarios/dashboard/tabs.cy.spec.js

```diff
@@ -91,45 +92,79 @@ describe("scenarios > dashboard > tabs", () => {
-    visitDashboardAndCreateTab({
-      dashboardId: ORDERS_DASHBOARD_ID,
-      save: false,
-    });
+  it(
+    "should allow moving dashcards between tabs",
+    { scrollBehavior: false },
+    () => {
+      visitDashboardAndCreateTab({
+        dashboardId: ORDERS_DASHBOARD_ID,
+        save: false,
+      });
 
-    goToTab("Tab 1");
+      goToTab("Tab 1");
 
-    cy.log("should stay on the same tab");
-    cy.findByRole("tab", { selected: true }).should("have.text", "Tab 1");
+      cy.log("add second card");
+      addTextBoxWhileEditing("Text card");
 
-    getDashboardCard(0).then(element => {
-      cy.wrap({
-        width: element.outerWidth(),
-        height: element.outerHeight(),
-      }).as("originalSize");
-    });
+      cy.log("should stay on the same tab");
+      cy.findByRole("tab", { selected: true }).should("have.text", "Tab 1");
 
-    cy.log("move card to second tab");
+      getDashboardCard(0).then(element => {
+        cy.wrap({
+          width: element.outerWidth(),
+          height: element.outerHeight(),
+        }).as("card1OriginalSize");
+      });
 
-    moveDashCardToTab({ tabName: "Tab 2" });
+      getDashboardCard(1).then(element => {
+        cy.wrap(element.offset()).as("card2OriginalPosition");
+      });
 
-    getDashboardCards().should("have.length", 0);
+      cy.log("move second card to second tab first, then the first card");
+      // moving the second card first to invert their position, this allows us
+      // to check if the position is restored when undoing the movement of the second one
+      moveDashCardToTab({ tabName: "Tab 2", dashcardIndex: 1 });
+      moveDashCardToTab({ tabName: "Tab 2", dashcardIndex: 0 });
 
-    goToTab("Tab 2");
+      cy.log("fist tab should be empty");
+      cy.findAllByTestId("toast-undo").should("have.length", 2);
+      getDashboardCards().should("have.length", 0);
 
-    getDashboardCards().should("have.length", 1);
+      cy.log("cards should be in second tab");
+      goToTab("Tab 2");
+      getDashboardCards().should("have.length", 2);
 
-    cy.log("size should stay the same");
+      cy.log("size should stay the same");
 
-    getDashboardCard(0).then(element => {
-      cy.get("@originalSize").then(originalSize => {
-        expect({
-          width: element.outerWidth(),
-          height: element.outerHeight(),
-        }).to.deep.eq(originalSize);
+      getDashboardCard(1).then(element => {
+        cy.get("@card1OriginalSize").then(originalSize => {
+          expect({
+            width: element.outerWidth(),
+            height: element.outerHeight(),
+          }).to.deep.eq(originalSize);
+        });
       });
-    });
-  });
+
+      cy.log("undoing movement of second card");
+
+      cy.findAllByTestId("toast-undo").eq(0).findByRole("button").click();
+
+      goToTab("Tab 1");
+
+      getDashboardCards().should("have.length", 1);
+
+      cy.log("second card should be in the original position");
```

Comment:
I'm not sure if this is acceptable, but the card won't always go back to the same location.

https://github.com/metabase/metabase/assets/6830683/b9755712-8ed2-48cd-a34d-4d11201f930b

Reply:
Given that it's the same behavior when undoing a card deletion I think it's fine, but maybe worth double checking with Connor/Dawei.

Reply:
We decided to add it to Milestone 3 of the epic